### DISCUSSION TITLE
Unable to create a visualizer, need help understanding

### DISCUSSION THREAD

**Discussion by weirdyang** [statelyai/xstate#2145] [Q&A] [answered]
Hi,

I'm not sure why this state machine I created does not work, 

https://xstate.js.org/viz/?gist=87c82e62f410fe8602b1656e371e2229

Did I design it incorrectly?

This is the error messag:

`3StateTree.js:112 Uncaught TypeError: Cannot read property 'stateNode' of undefined`

**Comment by ChrisShank** [comment, marked as answer]
I think that error in the visualizer was hiding this warning: `Warning: Compound state node 'client.active' has no initial state.`

**Reply by weirdyang** [reply]
Thank you, that fixed it. 

Now I'm stuck being unable to transition to the `connecting` state invoke's OnDone target. Any hint on what is wrong with my schema? Thanks again for your assistance.

**Reply by ChrisShank** [reply]
I'm not sure I fully understand what you are trying to model here, but there are a couple of things that don't *feel* right:
- The boolean flags in you context and the `disconnected` transition-less state don't seem necessary (to me). I think modeling the state chart differently could solve this.
- The ` on: { '': '.sendCommand' }` doesn't make sense to me. I also think modeling the state chart differently could solve this.
- The `disconnected` and `connected.idle` don't seem like they should be `type: 'final'`? 

Again not 100% sure what you are trying to model, but maybe this could be a good starting point?
https://xstate.js.org/viz/?gist=8fcc3c610d54d28ede0b46890aa53df5

**Reply by weirdyang** [reply]
Thanks a lot! I'm trying to learn state charts and I think I've got the concepts wrong. Will look at your example and the docs and try again

**Reply by ChrisShank** [reply]
Happy to answer any other questions, modeling is definitely the hardest part! 😁

**Reply by weirdyang** [reply]
Thanks! With your example and re-reading the docs and resources; I ended up with this; https://xstate.js.org/viz/?gist=62e0e2db0682df80dd3fd7de0afb4a12  

I realised the processing could also be a child state machine.

Thanks again!
